### ISSUE TITLE
Related models in the response not converting to json

### ISSUE THREAD

**Issue by ChristopherBrownie** [inertiajs/inertia-django#18]
How would we include related models through `select_related()` or `prefetch_related()` in the json output? The `model_to_dict()` method used in the encoder doesn't seem to include them. I'm realizing I almost need a DRF serializer lite setup. 

Is this something the default encoder could handle (instead of needing to make a custom one) or would it be easier for me just to use something like `values()` to select properties?

**Comment by BrandonShar** [comment]
This has been a sticking point for me as well and I've typically ended up writing custom serializers (or using values for simpler cases) in my project to handle it.

I would love for the default encoder to handle it, but I'm a bit worried about straying too far from expected Django behavior, so I've hesitated so far. For reference, this isn't an issue for the Rails or Laravel adapters because those frameworks have built in ways to handle json encoding models. 

How would your ideal api work here?

**Comment by mercuryseries** [comment]
For what it's worth, I personally use DRF and it works great.

**Comment by ChrisxHart** [comment]
I really like the simplicity of how Laravel handles [models to json serialization](https://laravel.com/docs/9.x/eloquent-serialization#relationships). For Django models that could look something like:
```python
class Model():
    class Meta:
        serialized_fields = ('id', 'value_field', 'related_model')
```
I was surprised Django doesn't support something like that out of the box. I asked around on the Django discord and got responses like "Django isn't a rest api framework" and "make your own serializer [json encoder]". In my opinion a web framework is a super set of a rest api framework, thus would expect to have it included.

I was hesitant to use DRF as it seems like overkill for what I would use it for, but maybe it's the best solution we have?
Something like [serpy](https://github.com/clarkduvall/serpy) seems more appropriate here as serializing is all I need it to do. This package may not be supported anymore though. Any other thoughts?

**Comment by BrandonShar** [comment]
I love the idea of that, but I think it would have enough edge cases that it's too complex for inertia to take on. I'm also on the same page where I think DRF is too complex for what I'm typically looking to do. 

Serpy looks pretty similar to what I've manually done in the past: I generally created pretty simple serializer dataclasses and then pass them instances of models.

It would be great if we could figure out a better solution, but I don't think it's worth it to create a serializer library within inertia. Maybe the ecosystem just needs a new serializer library in general!

@dankuck how are you currently handling serializing relationships in your project?

**Comment by dankuck** [comment]
When you were working on it you used serializers and we've kept up that tradition

**Comment by ChrisxHart** [comment]
So it sounds like we will seemingly always want to use a serializer library or convert the models to dictionaries ourselves. Maybe that could be stated in the django-vite doc somewhere, as data is treated the same way in adapters like laravel and RoR.
Not wanting to stray from Django behavior makes sense, the `model_to_dict()` behavior might already be doing that though. The `InertiaJsonEncoder` seems limited and borderline produces unexpected output in this case, whereas Django wouldn't encode any of it.

**Comment by BrandonShar** [comment]
Yeah that's a fair point. The `InertiaJsonEncoder` was intended to get people off the ground quickly, but maybe it's more hurt than help and it would be better to explicitly tell people they need to serialize on their own and let Django throw errors if they don't.

**Comment by danmanix** [comment]
Sooo, this is fairly hacky (and could prob be written more concisely), but this was a quick solution for me - rely on using prefetch_related in a prop QuerySet to hint which relations to traverse by adding to http.py:

```
 def deep_transform_callables(prop):

        if isinstance(prop, QuerySet):
            prefetches = prop._prefetch_related_lookups
            if prefetched:
                new_prop = []
                for item in prop:
                    new_item = model_to_dict(item)
                    for prefetch in prefetches:
                        if getattr(item, prefetch):
                            try:
                                new_item[prefetch] = getattr(item, prefetch)
                            except:
                                new_item[prefetch] = getattr(item, prefetch).all()
                    new_prop.append(new_item)
                return new_prop

        if not isinstance(prop, dict):
                ....
```

**Comment by vsg24** [comment]
This is bugging me as well, wish it was handled automatically

**Comment by BrandonShar** [comment]
#66 should handle this issue! (pending tag)
